src/iosrv/vos.c

```diff
@@ -646,10 +646,11 @@ close_iod(struct dss_enum_unpack_io *io)
 	D_ASSERTF(io->ui_iods_size <= io->ui_iods_cap, "%d < %d\n",
 		  io->ui_iods_size, io->ui_iods_cap);
 
-	if (io->ui_iods_size == 0)
+	if (io->ui_iods_size == 0 &&
+	    (io->ui_iods == NULL || io->ui_iods[0].iod_nr == 0))
 		return;
 
-	idx = io->ui_iods_size - 1;
+	idx = io->ui_iods_size == 0 ? 0 : io->ui_iods_size - 1;
```

Comment:
Don't quite follow this.

Reply:
This is to make sure the idx is not -1. The idx handling is not very clear indeed. I will try to clean it up in the next patch.